pkg/util/retry/retry.go

```diff
@@ -69,6 +73,18 @@ func Do(ctx context.Context, fn func() error, opts ...Option) error {
 
 // errUnrecoverable is error instance for unrecoverable.
 var errUnrecoverable = errors.New("unrecoverable error")
+var errContextDone = errors.New("context done error")
```

Comment:
maybe using `context.DeadlineExceeded` and `context.Canceled` will do
no need to introduce new error type here?